File: sdk/ocaml/lib/network_services.ml
(** Network Services: IP addresses and reverse DNS for services *)

(** GET /api/service/{id}/ip - List Service IP Addresses *)
let list_ips client params =
  HttpClient.get client "/api/service/{id}/ip" params

(** GET /api/service/{id}/rdns - Get reverse DNS entries *)
let get_rdns client params =
  HttpClient.get client "/api/service/{id}/rdns" params

(** POST /api/service/{id}/rdns - Update reverse DNS entries *)
let update_rdns client params =
  HttpClient.post client "/api/service/{id}/rdns" params

File: sdk/dart/lib/src/network_services.dart
import 'http_client.dart';

/// Network Services: IP addresses and reverse DNS for services
class NetworkServicesResource {
  final HttpClient httpClient;
  NetworkServicesResource(this.httpClient);

  /// GET /api/service/{id}/ip - List Service IP Addresses
  Future<Map<String, dynamic>> listIps([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/service/{id}/ip', params);
  }

  /// GET /api/service/{id}/rdns - Get reverse DNS entries
  Future<Map<String, dynamic>> getRdns([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/service/{id}/rdns', params);
  }

  /// POST /api/service/{id}/rdns - Update reverse DNS entries
  Future<Map<String, dynamic>> updateRdns([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/service/{id}/rdns', params);
  }

}
